### PR TITLE
[enhancement](tracing)Support forward to master tracing

### PR DESCRIPTION
**Pull request by Yukang-Lian**
# Proposed changes
## Problem summary

When the query statement is executed, some metadata processing requests will be sent to the follower fe, but the follower fe does not have execution permission, and it will be forwarded to the master fe for execution. This PR implements the tracing of forwarding information.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments
The effect is as shown in the figure.

FEMNAS_5%$(ZV`Z$M4PM`7M

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 1]
```diff
@@ -399,7 +413,17 @@ public void execute(TUniqueId queryId) throws Exception {
                                 + Env.getCurrentEnv().getSelfNode().first + ") and failed to execute"
                                 + " because Master FE is not ready. You may need to check FE's status");
                     }
-                    forwardToMaster();
+                    Span forwardToMasterSpan =
```
[the remark this answers is not in the document]
Comment (Yukang-LIAN) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:416]: Yeah, I got it.

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java [hunk 2]
```diff
@@ -534,6 +539,12 @@ public TMasterOpResult proxyExecute(TMasterOpRequest request) {
             }
         }
 
+        Map<String, String> traceCarrier = new HashMap<>();
+        if (request.isSetTraceCarrier()) {
+            traceCarrier = request.getTraceCarrier();
+        }
+        ctx.initTracer("master trace");
```
[the remark this answers is not in the document]
Comment (Yukang-LIAN) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:546]: Try to solve it.

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Yukang-LIAN) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java]: Done.

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Yukang-LIAN) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java]: Done.

**Comment by Yukang-LIAN** [comment]
This time The effect is as shown in the figure.

2}6P4)ZYSEWF$S{U{`L(GXQ

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java [hunk 3]
```diff
@@ -548,7 +561,21 @@ public TMasterOpResult proxyExecute(TMasterOpRequest request) {
                 UUID uuid = UUID.randomUUID();
                 queryId = new TUniqueId(uuid.getMostSignificantBits(), uuid.getLeastSignificantBits());
             }
-            executor.execute(queryId);
+            Context extractedContext = Telemetry.getOpenTelemetry().getPropagators().getTextMapPropagator()
```
Comment (luozenglin) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:564]: The extractedContext comes from the follower node and you should use `Span.fromContext(extractedContext).getSpanContext().isValid()` to check whether it is valid. If valid then initialize tracer.

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java [hunk 4]
```diff
@@ -548,7 +561,21 @@ public TMasterOpResult proxyExecute(TMasterOpRequest request) {
...
+                    .extract(Context.current(), traceCarrier, getter);
+            try (Scope extract = extractedContext.makeCurrent()) {
```
Comment (luozenglin) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:566]: Here it should not be `extractedContext.makeCurrent()`, but `masterQuerySpan.makeCurrent()`.

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java (hunk 3 above)
Comment (Yukang-LIAN) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java]: Done.

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java (hunk 4 above)
Comment (Yukang-LIAN) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:579]: Done.

**Comment by Yukang-LIAN** [comment]
I have tested that both `extractedContext.makeCurrent()` and `setParent(extractedContext)` work correctly when the `extractedContext` is invalid.

**Pull request closed without merging**